crates/sui-core/src/gateway_types.rs

```diff
@@ -321,7 +329,7 @@ pub struct SuiParsedMoveObject {
 }
 
 impl SuiMoveObject for SuiParsedMoveObject {
-    fn try_from(
+    fn try_from_layout(
```

Comment:
I know it's not from this diff, but can we require `MoveStructLayout` in this func rather than `Option<MoveStructLayout>`?

Reply:
I have changed this in `SuiMoveObject`, but we still have the same in `SuiObject` and `SuiObjectRead`, the root cause is because we include the layout in `ObjectRead` which is Optional because layout is not applicable to Package, I will do another refactoring in subsequence PR to remove layout from `ObjectRead` and solely use module resolver instead.